Ops/src/OpsWorkerGenetic.cpp

```diff
@@ -147,66 +143,91 @@ void WorkerGenetic::setMutationProbability(double value)
         }
     }
-
-    bestValue = std::numeric_limits<double>::max();
-    for(size_t i=0; i<mNumPoints; ++i) {
-        double value = mObjectives[i]*opsRand();
-        if(value < bestValue && i != mParent1) {
-            mParent2 = i;
-            bestValue = value;
+    int bestId = sortedIds[bestCompetitor];
+    mParent1 = bestId;
+
+    // Select parent 2
+    competitors.clear();
+    competitors.push_back(opsRand()*(mNumPoints));
+    competitors.push_back(opsRand()*(mNumPoints));
+    while(competitors.size() < mObjectives.size()/3.0) {
+        competitors.push_back(opsRand()*(mNumPoints));
+    }
+    bestCompetitor = 0;
+    for(auto competitor : competitors) {
+        if(competitor > bestCompetitor) {
+            bestCompetitor = competitor;
         }
     }
+    bestId = sortedIds[bestCompetitor];
+    mParent2 = bestId;
 }
 
 
 //! @brief Auxiliary function for perform crossover
 void WorkerGenetic::crossOver()
 {
-    for(size_t i=0; i<mNumParameters; ++i)
+    mCandidatePoints.push_back(mPoints[mParent1]);
+    mCandidatePoints.push_back(mPoints[mParent2]);
+    double doCrossOver = opsRand();
+    if(doCrossOver < mCrossoverProbability)
     {
-        double x1 = mPoints[mParent1][i];
-        double x2 = mPoints[mParent2][i];
 
-        double doCrossOver = opsRand();
-        if(doCrossOver > mCrossoverProbability)
-        {
-            mChild1[i] = x1;
-            mChild2[i] = x2;
-        }
-        else
+        for(size_t i=0; i<mNumParameters; ++i)
         {
+            double x1 = mPoints[mParent1][i];
+            double x2 = mPoints[mParent2][i];
+
             double mean = (x1+x2)/2.0;
-            double dev = fabs(x2-x1)/2.0;
+            double dev = fabs(x2-x1)/10.0;
 
-            mChild1[i] = gaussian(mean,dev,mParameterMin[i],mParameterMax[i]);
-            mChild2[i] = gaussian(mean,dev,mParameterMin[i],mParameterMax[i]);
+            mCandidatePoints[mCandidatePoints.size()-2][i] = gaussian(mean,dev,mParameterMin[i],mParameterMax[i]);
+            mCandidatePoints[mCandidatePoints.size()-1][i] = gaussian(mean,dev,mParameterMin[i],mParameterMax[i]);
         }
     }
-    mCandidatePoints[mParent1] = mChild1;
-    mCandidatePoints[mParent2] = mChild2;
 }
 
 
 //! @brief Auxiliary function for mutating children
 void WorkerGenetic::mutate()
 {
+    std::vector<size_t> eliteIds = getIdsSortedFromWorstToBest();
+    while(eliteIds.size() > mNumElites) {
+        eliteIds.erase(eliteIds.begin());
+    }
     for(size_t i=0; i<mNumPoints; ++i) {
+        if(std::find(eliteIds.begin(),eliteIds.end(),i) != eliteIds.end()) {
+            continue;
+        }
         for(size_t j=0; j<mNumParameters; ++j) {
             double doMutation = opsRand();
             if(doMutation < mMutationProbability) {
                 double mean = mCandidatePoints[i][j];
-                double dev = (mParameterMax[j]-mParameterMin[j])/2.0;
+                double dev = (mParameterMax[j]-mParameterMin[j])/100.0;
```

Comment:
Again, unclear (for me) magic value.

Reply:
Same as above, it makes the distribution for mutation more narrow around the average value.

Reply:
OK, but should probably be tunable